parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -303,24 +303,26 @@ public void startColumn(ColumnDescriptor descriptor,
    * writes a dictionary page page
    * @param dictionaryPage the dictionary page
    */
-  public void writeDictionaryPage(DictionaryPage dictionaryPage) throws IOException {
+  public PageHeaderWithOffset writeDictionaryPage(DictionaryPage dictionaryPage) throws IOException {
     state = state.write();
     if (DEBUG) LOG.debug(out.getPos() + ": write dictionary page: " + dictionaryPage.getDictionarySize() + " values");
     currentChunkDictionaryPageOffset = out.getPos();
     int uncompressedSize = dictionaryPage.getUncompressedSize();
     int compressedPageSize = (int)dictionaryPage.getBytes().size(); // TODO: fix casts
-    metadataConverter.writeDictionaryPageHeader(
+    final PageHeader pageHeader = metadataConverter.writeDictionaryPageHeader(
         uncompressedSize,
         compressedPageSize,
         dictionaryPage.getDictionarySize(),
         dictionaryPage.getEncoding(),
         out);
-    long headerSize = out.getPos() - currentChunkDictionaryPageOffset;
+    long currentPos = out.getPos();
+    long headerSize = currentPos - currentChunkDictionaryPageOffset;
     this.uncompressedLength += uncompressedSize + headerSize;
     this.compressedLength += compressedPageSize + headerSize;
     if (DEBUG) LOG.debug(out.getPos() + ": write dictionary page content " + compressedPageSize);
     dictionaryPage.getBytes().writeAllTo(out);
     currentEncodings.add(dictionaryPage.getEncoding());
+    return new PageHeaderWithOffset(pageHeader, currentPos); // dictionary data starts from currentPos
```

Comment:
why don't we have the headerSize too? you should remove the 2-param constructor of PageHeaderWithOffset